diff --git a/haskell/private/actions/compile.bzl b/haskell/private/actions/compile.bzl
index c4ddb7a..5efd8fb 100644
--- a/haskell/private/actions/compile.bzl
+++ b/haskell/private/actions/compile.bzl
@@ -272,6 +272,7 @@ def _compilation_defaults(hs, cc, java, dep_info, plugin_dep_info, cc_info, srcs
         "-v0",
         "-no-link",
         "-fPIC",
+        "-fexternal-dynamic-refs",
         "-hide-all-packages",
         # Should never trigger in sandboxed builds, but can be useful
         # to debug issues in non-sandboxed builds.
diff --git a/haskell/private/haskell_impl.bzl b/haskell/private/haskell_impl.bzl
index 701c019..17360e7 100644
--- a/haskell/private/haskell_impl.bzl
+++ b/haskell/private/haskell_impl.bzl
@@ -552,7 +552,7 @@ def haskell_library_impl(ctx):
         actions = ctx.actions,
         feature_configuration = feature_configuration,
         dynamic_library = dynamic_library,
-        static_library = static_library,
+        pic_static_library = static_library,
         cc_toolchain = cc_toolchain,
     )
     compilation_context = cc_common.create_compilation_context()
